CODINGSTYLE.md

```diff
@@ -0,0 +1,56 @@
+# Coding Style
+
```

Comment:
I remove the introduction, and modify the filename to [CODINGSTYLE.md](https://github.com/solvcon/modmesh/pull/362/files#diff-afde3672b6b3ae60dabe50d33a0d268401dbb5cfe93dd164bbdac6284f293f8f).